### PR TITLE
[FIX] website_product_comment_purchased: Fix performance issue when create a mail_message

### PR DESCRIPTION
**Pull request by hugho-ad**
Fix https://github.com/Vauxoo/yoytec/issues/716
Fix https://github.com/Vauxoo/yoytec/issues/671

### CODE REVIEW

**Comment by hbto** [comment]
@hugho-ad 

do not forget to use a for cycle 

For the ids and not call the ids[0]

**Comment by hugho-ad** [comment]
@hbto got it

File: website_product_comment_purchased/models/mail.py [hunk 1]
```diff
@@ -22,37 +22,38 @@
...
+        res = {}.fromkeys(ids, False)
         sale_report_obj = self.pool.get('sale.report')
-        product_obj = self.pool.get('product.product')
-        cr.execute("""
-            {0}
-            FROM ( {1} )
-            {2}
-            """.format(sale_report_obj._select(), sale_report_obj._from(),
-                       sale_report_obj._group_by()))
         mail_cache = self.browse(cr, uid, ids, context)[0]
```
Comment (moylop260) [inline website_product_comment_purchased/models/mail.py]: You can avoid get a browse of all ids if you will get just the first item with:
`mail_cache = self.browse(cr, uid, ids[0], context)`

File: website_product_comment_purchased/models/mail.py [hunk 2]
```diff
@@ -22,37 +22,38 @@
...
+            FROM ( {_from} )
+            {where}
+            {group}
+            """.format(
+            select=sale_report_obj._select(),
+            _from=sale_report_obj._from(),
+            where=where,
+            group=sale_report_obj._group_by())
+        cr.execute(execute)
+        if cr.fetchall():
```
Comment (moylop260) [inline website_product_comment_purchased/models/mail.py]: If you need a boolean then you could avoid get all rows.
You can use `LIMIT 1` in sql and a `cr.fetchone()` in py.

**Comment by moylop260** [comment]
@hugho-ad 
I added small comments.

Could you share us the result of performance before of this PR and after of this pr, please?

**Comment by moylop260** [comment]
:+1:

**Comment by moylop260** [comment]
@nhomar 
Are you agree with this change?

**Comment by moylop260** [comment]
fix
@hugho-ad 
This pr fix 167 too?
Joke!

**Comment by moylop260** [comment]
I apply this fix to last build of legacy and work very good!
If exists a new comment we can create a new pr.

**Pull request merged**